cdap-app-templates/cdap-etl/cdap-etl-api/src/main/java/io/cdap/cdap/etl/api/aggregation/WindowAggregationDefinition.java

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright © 2022 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.etl.api.aggregation;
+
+import io.cdap.cdap.etl.api.relational.Expression;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Specifies how a window aggregation operation should be executed.
+ */
+public class WindowAggregationDefinition extends AggregationDefinition {
+  List<Expression> partitionExpressions;
```

Comment:
The window expression should also include an order by expression and a window frame definition. Essentially everything specified as an input in the plugin should be a part of the definition.

Screenshot for reference: 
![image](https://user-images.githubusercontent.com/1991785/193266301-9c7127a6-7744-4f0e-ab63-bc1e32d920c9.png)

Reply:
Also is there a specific reason `partitionExpressions` isn't `private`?

Reply:
+1 here, we should have all the properties needed to define a window aggregation function in this PR

Reply:
Added all properties